### DISCUSSION TITLE
OTP/PIN Field

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1352] [Proposals]
### Goal

Avaliar proposta de um novo componente para ser adicionado no Mística.

### Context

Necessitamos da criação de um novo componente para aplicar de forma consistente e reutilizável em alguns fluxos, como: Login, Cadastro, Pay (finanças), entre outros.

Criamos esses componentes na Lib Intermediária para suprir a necessidade dos Designers, mas caso esse componente for ficar alocado no Mística, podemos depreciar e fazer as alterações necessárias.

### Impact

Adicionar esse componente reduziria o tempo de desenvolvimento nas squads e teríamos consistência em sua aplicação nos fluxos mencionados acima.

### Discussion Questions

ND

### Do you have a visual proposal?

https://www.figma.com/file/7c8RdB62AdlAgc8sogquRX/OTP%2FPIN-Field?type=design&node-id=0%3A1&mode=design&t=SxibTM8ANGghMlpp-1

**Comment by aweell** [comment]
Hi @annefranca,

There's already a request from another team to include this component in Mística https://github.com/Telefonica/mistica-design/issues/931

In this file, there are early explorations of how the component would look like:

https://www.figma.com/file/e8d3PkXitUJ1Jwgc9Y3mNc/branch/tZI4DDpfEqWJNlKNCNrlMM/%F0%9F%94%B8-Input-Fields-%26-Forms-Component-Specs?type=design&node-id=1447-8501&mode=design&t=GQhFVoKnudObPWBX-0

**Reply by annefranca** [reply]
Olá @aweell 

Nesse arquivo tem o estudo do PIN, certo? Poderia incluir o OTP também?

**Reply by AnaMontes11** [reply]
Hi @annefranca ,

We believe it makes sense for OTP and PIN to be a single component, so we need to work on that and think about how it would function for both cases. We would like to hear your thoughts on how you believe the component should function on the PIN entry screen and its inverse version.

Looking forward to your response.

![MicrosoftTeams-image (27)](https://github.com/Telefonica/mistica-design/assets/88187691/63b4a4ea-1d51-4390-ae63-d7cce37d0df4)

**Reply by annefranca** [reply]
Obrigada @AnaMontes11  👍

Vamos avaliar com a equipe e o mais breve possível voltamos aqui com um direcionamento.

@vinrosa @BrunoBiagioniNeto alguma consideração?

**Reply by vinrosa** [reply]
Minha única observação sobre a diferença entre o PIN e o OTP é que:

PIN tem um comportamento mais próximo de um input do tipo PASSWORD
OTP tem um comportamento mais próximo de um input do tipo TEXT

**Reply by annefranca** [reply]
@AnaMontes11 @yceballost

**Reply by aweell** [reply]
Hello @vinrosa, @annefranca 

The base component can be the same.

> PIN tem um comportamento mais próximo de um input do tipo PASSWORD
> OTP tem um comportamento mais próximo de um input do tipo TEXT

IMO the most important difference based on the design provided is that the PIN values have a mask and OTP values don't. The other elements needed to build both components are shared (despite the fact that the input design is different) but if we want to unify both it should be possible (and beneficial):

* Provide a way to define the length of the password/pin/OTP (since it can vary from 4 to 6 numbers)
* Provide a way to define if is masked or not (masked for PIN / unmasked for OTP)

Here's an example of the boolean mask property that Uber uses in their design system PIN Code component:

https://baseweb.design/components/pin-code/#adding-mask

**Reply by annefranca** [reply]
@aweell você consegue fazer uma previsão visual desses cenários e compartilhar com a gente?

**Reply by aweell** [reply]
Sure!

![Screenshot 2023-07-21 at 13 33 18](https://github.com/Telefonica/mistica-design/assets/44420072/1d38d119-97ab-444f-923b-1c922fb386ec)

**Reply by annefranca** [reply]
Perfeito, @aweell 

Funciona na nossa aplicação.

Quais seriam os próximos passos?

**Comment by aweell** [comment]
There's a branch created for the component specs with Vivo Design Ops with edition permissions: https://www.figma.com/file/e8d3PkXitUJ1Jwgc9Y3mNc/branch/r2ZlMD7ctYzJ6XGYIcqfur/%F0%9F%94%B8-Input-Fields-%26-Forms-Component-Specs?type=design&node-id=2388%3A7765&mode=design&t=cfvgHrZfODVITqlV-1

Vivo can work in the definition of the component to include it as core component.

I close this discussion and I link it to the issue ticket #931
